lib/src/tooltip/Tooltip.test.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { render, waitFor } from "@testing-library/react";
+import DxcTooltip from "./Tooltip";
+import DxcButton from "../button/Button";
+import userEvent from "@testing-library/user-event";
+
+(global as any).globalThis = global;
+(global as any).DOMRect = {
+  fromRect: () => ({ top: 0, left: 0, bottom: 0, right: 0, width: 0, height: 0, x: 0, y: 0 }),
+};
+(global as any).ResizeObserver = class ResizeObserver {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+};
+
+describe("Tooltip component tests", () => {
+  test("Tooltip does not render by default", async () => {
+    const { queryByRole } = render(
+      <DxcTooltip label="Tooltip Test">
+        <DxcButton label="Hoverable button" />
+      </DxcTooltip>
+    );
+    await waitFor(() => {
+      const tooltipElement = queryByRole("tooltip");
+      expect(tooltipElement).toBeFalsy();
+    });
+  });
+
+  test("Tooltip renders with correct label on hover", async () => {
+    const { getByText, getByRole } = render(
+      <DxcTooltip label="Tooltip Test">
+        <DxcButton label="Hoverable button" />
+      </DxcTooltip>
+    );
+    const triggerElement = getByText("Hoverable button");
+    await userEvent.hover(triggerElement);
+    await waitFor(() => {
+      const tooltipContent = getByRole("tooltip", { name: "Tooltip Test" });
+      expect(tooltipContent).toBeTruthy();
+    });
+  });
+
+  test("Tooltip stops being rendered when hover is stopped", async () => {
```

Comment:
the first 3 tests can be merged into a single test that mimics better the user behavior.

Reply:
Should we also test the delay since it is a value we are assigning?

Reply:
I am currently having trouble setting it up because the code from the event `unhover` seems unreachable after a `waitFor`, will probably need some help to fix it.

Reply:
The current code that I am testing is:

```
test("Tooltip displays when intended", async () => {
    const { getByText, queryByRole, getByRole } = render(
      <DxcTooltip label="Tooltip Test">
        <DxcButton label="Hoverable button" />
      </DxcTooltip>
    );
    const triggerElement = getByText("Hoverable button");
    await waitFor(() => {
      const tooltipElement = queryByRole("tooltip");
      expect(tooltipElement).toBeFalsy();
    });
    await userEvent.hover(triggerElement);
    await waitFor(() => {
      const tooltipContent = getByRole("tooltip", { name: "Tooltip Test" });
      expect(tooltipContent).toBeTruthy();
    });
    await userEvent.unhover(triggerElement);
    await waitFor(() => {
      const tooltipElement = queryByRole("tooltip");
      expect(tooltipElement).toBeFalsy();
    });
  });
```

![image](https://github.com/dxc-technology/halstack-react/assets/33158831/39af205b-200f-4cba-b4f9-59e155c8a4c6)